### PR TITLE
Upgrade project to RN61 with CocoaPods

### PR DESCRIPTION
**Pull request by LeoNatan**
- Remove componentWillMount functions
- Remove PushNotificationIOS
- Add pod install step to lerna bootstrap

**Todo:**
- [ ] Figure out CI
- [ ] Remove iOS sample projects from CI (no desire to maintain them with CocoaPods or at all)

### CODE REVIEW

File: scripts/install.ios.sh [hunk 1]
```diff
@@ -2,6 +2,11 @@
...
+gem install xcpretty >/dev/null 2>&1
+export GEM_HOME=$HOME/.gem
+export PATH=$GEM_HOME/bin:$PATH
+HOMEBREW_NO_INSTALL_CLEANUP=1 HOMEBREW_NO_AUTO_UPDATE=1 brew install cocoapods
+
 export CODE_SIGNING_REQUIRED=NO
-brew tap wix/brew
-brew install applesimutils --HEAD
\ No newline at end of file
+HOMEBREW_NO_INSTALL_CLEANUP=1 HOMEBREW_NO_AUTO_UPDATE=1 brew tap wix/brew
```
Comment (EvanBacon) [inline scripts/install.ios.sh]: 👍 nice

**Comment by msand** [comment]
It seems like the latest version of Detox works fine with v0.61.4 of react-native as well: https://github.com/msand/react-native-svg-e2e/pull/13

**Comment by LeoNatan** [comment]
This PR is only bureaucracy to get the test project and CI to work with 61.x. Detox on iOS should support RN 61 without issues as far as I know.

**Comment by msand** [comment]
Any known issues on android? At least building and taking screenshots of all the tests in the rnsvg-e2e repo seem to work correctly.

**Comment by LeoNatan** [comment]
Android is tracked here: https://github.com/wix/Detox/issues/1731

**Comment by LeoNatan** [comment]
@rotemmiz Reminder. Thanks

**Comment by rotemmiz** [comment]
You're welcome @LeoNatan 😗

**Comment by LeoNatan** [comment]
It just werks???

**Comment by d4vidi** [comment]
Somehow the builds are red now - with stuff related to pods, @LeoNatan could you check it out plz?

**Comment by LeoNatan** [comment]
@d4vidi Sure

**Comment by LeoNatan** [comment]
@d4vidi The error is caused by postinstall somehow:

> 08:06:53 lerna ERR! execute Error occured with 'undefined' while running 'npm run postinstall'

And since cocoa pods looks for its dependencies in node_modules/, it fails.

**Pull request merged**